samples/performance/image_compression_control/image_compression_control.cpp

```diff
@@ -0,0 +1,588 @@
+			                                                                          color_image_info.tiling,
+			                                                                          color_image_info.usage,
+			                                                                          0,        // no create flags
+			                                                                          &format_properties);
+
+			// Pick the first format that supports at least 2 or more levels of fixed-rate compression, otherwise
+			// pick the first format that supports at least 1 level
+			if (result != VK_ERROR_FORMAT_NOT_SUPPORTED &&
+			    fixed_rate_flags.size() > 0)
+			{
+				if (VK_FORMAT_UNDEFINED == chosen_format)
+				{
+					chosen_format                    = candidate_format;
+					supported_fixed_rate_flags_color = fixed_rate_flags;
+				}
+
+				if (fixed_rate_flags.size() > 1)
+				{
+					chosen_format                    = candidate_format;
+					supported_fixed_rate_flags_color = fixed_rate_flags;
+					break;
+				}
+			}
+		}
+
+		// Fallback if no fixed-rate compressible format was found
+		color_image_info.format = (VK_FORMAT_UNDEFINED != chosen_format) ? chosen_format : swapchain_image.get_format();
+		LOGI("Chosen color format: {}", vkb::to_string(color_image_info.format));
+
+		// Hide GUI fixed-rate compression option if chosen format does not support it
+		if (supported_fixed_rate_flags_color.empty())
+		{
+			gui_skip_compression_values.insert(TargetCompression::FixedRate);
+
+			LOGW("Color image does not support fixed-rate compression. Possible reasons:");
+			LOGW("\t- Its format may not be supported (format = {})", vkb::to_string(color_image_info.format));
+			if (color_image_info.usage & VK_IMAGE_USAGE_STORAGE_BIT)
+			{
+				LOGW("\t- It is a storage image (usage = {})", vkb::image_usage_to_string(color_image_info.usage));
+			}
+			if (color_image_info.samples > 1)
+			{
+				LOGW("\t- It is a multi-sampled image (sample count = {})", vkb::to_string(color_image_info.samples));
+			}
+		}
+	}
+
+	vkb::core::Image depth_image{*device,
+	                             vkb::core::ImageBuilder(color_image_info.extent)
+	                                 .with_format(vkb::get_suitable_depth_format(device->get_gpu().get_handle()))
+	                                 .with_usage(VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT | VK_IMAGE_USAGE_TRANSIENT_ATTACHMENT_BIT)
+	                                 .with_vma_usage(VMA_MEMORY_USAGE_GPU_ONLY)};
+
+	vkb::core::ImageBuilder color_image_builder(color_image_info.extent);
+	color_image_builder.with_format(color_image_info.format)
+	    .with_usage(color_image_info.usage)
+	    .with_tiling(color_image_info.tiling)
+	    .with_vma_usage(VMA_MEMORY_USAGE_GPU_ONLY);
+
+	// Prepare compression control structure
+	VkImageCompressionFixedRateFlagsEXT color_fixed_rate_flags[1] = {compression_fixed_rate_flag_color};
```

Comment:
I prefer to leave them separate (no helper) since compressing the image and the surface are two different use cases. I simplified the code as you suggested.